src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs

```diff
@@ -30,10 +31,10 @@ public sealed partial class JsonSerializerOptions
         [RequiresUnreferencedCode(JsonSerializer.SerializationUnreferencedCodeMessage)]
         private void RootBuiltInConverters()
         {
-            if (s_defaultSimpleConverters is null)
+            if (Volatile.Read(ref s_defaultFactoryConverters) is null)
```

Comment:
I know this could not be a direct backport. So I assume the comments in the `main` change do not apply here? https://github.com/dotnet/runtime/pull/66522/files#diff-143351693c2cb653ad82e34086cfd8863b0bda66c8b498ed8a400924c4f5224e

Reply:
They do, but this is fairly standard use of Volatile so I don't believe it warrants explanation.